sql/core/src/test/resources/sql-tests/inputs/pgSQL/int2.sql

```diff
@@ -0,0 +1,129 @@
+--
+-- Portions Copyright (c) 1996-2019, PostgreSQL Global Development Group
+--
+--
+-- INT2
+-- https://github.com/postgres/postgres/blob/REL_12_BETA1/src/test/regress/sql/int2.sql
+
+CREATE TABLE INT2_TBL(f1 smallint) USING parquet;
+
+-- [SPARK-28023] Trim the string when cast string type to other types
+INSERT INTO INT2_TBL VALUES (trim('0   '));
```

Comment:
It seems that we had better merge the SPARK-28023 PR before merging this.
cc @gatorsmile